source/MRMesh/MRPointsSave.cpp

```diff
@@ -64,28 +67,32 @@ tl::expected<void, std::string> toPly( const PointCloud& points, std::ostream& o
             cVert.p = points.points[VertId( v )];
             const auto& c = ( *colors )[VertId( v )];
             cVert.r = c.r; cVert.g = c.g; cVert.b = c.b;
-            out.write( (const char*) &cVert, 15 );
+            out.write( ( const char* )&cVert, 15 );
+            if ( callback && !callback( float( v ) / numVertices ) )
```

Comment:
again too expensive to invoke callback on every triangle

Reply:
fixed